index.php

```diff
@@ -43,27 +43,33 @@
 
 // We are going to need lots of memory and time.
 raise_memory_limit(MEMORY_HUGE);
-set_time_limit(300);
+set_time_limit(600);
 
 $mform = new local_codechecker_form(new moodle_url('/local/codechecker/'));
 $mform->set_data((object)$pageparams);
 if ($data = $mform->get_data()) {
     redirect(new moodle_url('/local/codechecker/', $pageparams));
 }
 
-if ($path) {
-    $fullpath = $CFG->dirroot . '/' . trim($path, '/');
-    if (!is_file($fullpath) && !is_dir($fullpath)) {
-        $fullpath = null;
-    }
-}
-
 $output = $PAGE->get_renderer('local_codechecker');
 
 echo $OUTPUT->header();
 
-if ($path) {
-    if ($fullpath) {
+if ($pathlist) {
+    $paths = preg_split('~[\r\n]+~', $pathlist);
+
+    $ok = true;
+    $fullpaths = [];
+    foreach ($paths as $path) {
+        $fullpath = $CFG->dirroot . '/' . trim($path, '/');
+        if (!is_file($fullpath) && !is_dir($fullpath)) {
+            $fullpaths = [];
+            break;
```

Comment:
More yet, and related with the above, why break? Wouldn't be better to just accumulate the correct ones and skip the non-existing ones?